core/arch/aarch64/emit_utils.c

```diff
@@ -947,16 +956,23 @@ emit_indirect_branch_lookup(dcontext_t *dc, generated_code_t *code, byte *pc,
         APP(&ilist, INSTR_CREATE_b(dc, opnd_create_instr(load_tag)));
     }
 
-    APP(&ilist, miss);
-    /* Recover the dcontext->last_exit to x0 */
-    APP(&ilist, instr_create_restore_from_tls(dc, DR_REG_R0, TLS_REG3_SLOT));
-
     /* Target delete entry */
     APP(&ilist, target_delete_entry);
     add_patch_marker(patch, target_delete_entry, PATCH_ASSEMBLE_ABSOLUTE,
                      0 /* beginning of instruction */,
                      (ptr_uint_t *)&ibl_code->target_delete_entry);
 
+    /* Get back next_tag's value previously saved in r1 by lookup code. This will be
+     * stored in dcontext below.
+     */
+    APP(&ilist,
+        XINST_CREATE_move(dc, opnd_create_reg(DR_REG_R2), opnd_create_reg(DR_REG_R1)));
+
+    APP(&ilist, miss);
+
+    /* Recover the dcontext->last_exit to x0 */
+    APP(&ilist, instr_create_restore_from_tls(dc, DR_REG_R0, TLS_REG3_SLOT));
```

Comment:
I see there's a special linkstub `linkstub_ibl_deleted` that's used by x86 code for cases [when linkstub isn't present](https://github.com/DynamoRIO/dynamorio/blob/e5ac96b4886785546d4b02a1dbcdd132d0c1954b/core/arch/x86/emit_utils.c#L2687). But here as we do have the last linkstub saved in `TLS_REG3_SLOT`. I'm just restoring that.

Reply:
x86 actually never uses a linkstub for indirect branches (-indirect_stubs is off; exit cti goes straight to IBL) but aarchxx with reachability problems will probably always have a stub.

Reply:
I meant what linkstub address is set in `dcontext->last_exit`. I see that on AArch64, `ibl-stress` passes no matter if I set it to the actual last linkstub or `linkstub_ibl_deleted`. But as the former is available here, I use that.

Reply:
What are the consequences of having a real instead of special linkstub?  Without the special linkstub, the stat num_exits_ibl_deleted won't be incremented.  Are there any asserts or warnings about not knowing why DR exited the cache?

Reply:
Didn't find any relevant asserts or warnings when using the special linkstub. Made the change to use it, and I see proper stats for `Fcache exits, deleted but hit in ibl` now.